### PR TITLE
automatically deploy binary releases to github releases upon tag

### PR DESCRIPTION
**Pull request by patflynn**
I couldn't figure out a way to test this locally so there may be some trial and error cutting releases to make sure this works.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,4 +6,13 @@ before_cache:
 cache:
   directories:
     - $HOME/.gradle/caches/
-    - $HOME/.gradle/wrapper/
\ No newline at end of file
+    - $HOME/.gradle/wrapper/
+deploy:
```
Comment (loosebazooka) [inline .travis.yml]: is there any way you can explain this or point to docs?

Comment (patflynn) [inline .travis.yml]: https://docs.travis-ci.com/user/deployment/releases/

**Comment by patflynn** [comment]
related to #245

**Comment by loosebazooka** [comment]
lgtm

**Pull request merged**